peekingduck/pipeline/nodes/model/jde_mot/jde_files/multitracker.py

```diff
@@ -0,0 +1,546 @@
+            )
+            for i, (mean, cov) in enumerate(zip(multi_mean, multi_covariance)):
+                stracks[i].mean = mean
+                stracks[i].covariance = cov
+
+    def activate(self, kalman_filter: KalmanFilter, frame_id: int) -> None:
+        """Start a new tracklet.
+
+        Args:
+            kalman_filter (KalmanFilter): Kalman filter for state estimation.
+            frame_id (int): ID for current frame.
+        """
+        self.kalman_filter = kalman_filter
+        self.track_id = self.next_id()
+        self.mean, self.covariance = self.kalman_filter.initiate(
+            self.tlwh_to_xyah(self._tlwh)
+        )
+
+        self.tracklet_len = 0
+        self.state = TrackState.Tracked
+        # self.is_activated = True
+        self.frame_id = frame_id
+        self.start_frame = frame_id
+
+    def re_activate(
+        self, new_track: List[Any], frame_id: int, new_id: bool = False
+    ) -> None:
+        """Re-activate STrack.
+
+        Args:
+            new_track (List[Any]): List of new STracks.
+            frame_id (int): ID for current frame.
+            new_id (bool, optional): New track ID. Defaults to False.
+        """
+        self.mean, self.covariance = self.kalman_filter.update(
+            self.mean, self.covariance, self.tlwh_to_xyah(new_track.tlwh)
+        )
+
+        self.update_features(new_track.curr_feat)
+        self.tracklet_len = 0
+        self.state = TrackState.Tracked
+        self.is_activated = True
+        self.frame_id = frame_id
+        if new_id:
+            self.track_id = self.next_id()
+
+    def update(
+        self, new_track: List[Any], frame_id: int, update_feature: bool = True
+    ) -> None:
+        """Update a matched track.
+
+        Args:
+            new_track (List[STrack]): New STrack.
+            frame_id (int): Frame ID.
+            update_feature (bool, optional): Update feature. Defaults to True.
+        """
+        self.frame_id = frame_id
+        self.tracklet_len += 1
+
+        new_tlwh = new_track.tlwh
+        self.mean, self.covariance = self.kalman_filter.update(
+            self.mean, self.covariance, self.tlwh_to_xyah(new_tlwh)
+        )
+        self.state = TrackState.Tracked
+        self.is_activated = True
+
+        self.score = new_track.score
+        if update_feature:
+            self.update_features(new_track.curr_feat)
+
+    @property
+    def tlwh(self) -> np.ndarray:
+        """Get current position in bounding box format.
+
+        Returns:
+            np.ndarray: `(top left x, top left y, width, height)`.
+        """
+        if self.mean is None:
+            return self._tlwh.copy()
+        ret = self.mean[:4].copy()
+        ret[2] *= ret[3]
+        ret[:2] -= ret[2:] / 2
+        return ret
+
+    @property
+    def tlbr(self) -> np.ndarray:
+        """Convert bounding box to format `(min x, min y, max x, max y)`.
+
+        Returns:
+            np.ndarray: `(top left, bottom right)`.
```

Comment:
split up x and y instead, this makes it look like we are returning 2 elements

Reply:
Updated accordingly